### PR TITLE
8266250: WebSocketTest and WebSocketProxyTest call assertEquals(List<byte[]>, List<byte[]>)

### PR DESCRIPTION
**Pull request by dfuch**
Please find here an almost trivial test fix that should improve diagnostic in case of failures. 
It also avoids relying on an unspecified behavior of `Assert.assertEquals`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266250](https://bugs.openjdk.java.net/browse/JDK-8266250): WebSocketTest and WebSocketProxyTest call assertEquals(List<byte[]>, List<byte[]>)

### Reviewers
 * [Pavel Rappo](https://openjdk.java.net/census#prappo) (@pavelrappo - **Reviewer**) ⚠️ Review applies to 22dc80600672ffdd1cb8555439363bb47e5eac36

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3776/head:pull/3776` \
`$ git checkout pull/3776`

Update a local copy of the PR: \
`$ git checkout pull/3776` \
`$ git pull https://git.openjdk.java.net/jdk pull/3776/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3776`

View PR using the GUI difftool: \
`$ git pr show -t 3776`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3776.diff">https://git.openjdk.java.net/jdk/pull/3776.diff</a>

</details>

### CODE REVIEW

**Comment by pavelrappo** [comment]
> It also avoids relying on an unspecified behavior of `Assert.assertEquals`.

Which behavior is that? If I recall correctly, the test was written for TestNG 6.9.5, which provides a [method to compare collections](https://github.com/cbeust/testng/blob/ef2d1199abff4e1b8fa4b1148c1314e776d7a044/src/main/java/org/testng/Assert.java#L508-L517).

**Comment by pavelrappo** [comment]
> Exactly - that API specification doesn't say whether elements
> that are arrays will be compared using shallow equals or deep
> equals.

Fair enough.

Since you are here, consider further improving diagnosability. The failed comparison should provide more context. Given that the comparables are relatively small, it makes sense to print them out in full. Currently, `assertEquals` short-circuits by reporting only one property of the lists that differs. If the lists differ in size, it's the size that is reported. If the lists are of the same size, it's the leftmost non-equal element that is reported, if any.

**Comment by pavelrappo** [comment]
> Exactly - that API specification doesn't say whether elements
> that are arrays will be compared using shallow equals or deep
> equals.

Fair enough.

Since you are here, consider further improving diagnosability. The failed comparison should provide more info. 
Currently, assertEquals short-circuits by reporting only one property of the lists that differs. If the lists are of different size, it's the size that is reported. If the lists are of the same size, it's the leftmost non-equal element that is reported, if any. In either case, there might be a lot of info missing. Given that the comparables are relatively small, it makes sense to print them out in full.

**Comment by dfuch** [comment]
@pavelrappo: done.

File: test/jdk/java/net/httpclient/websocket/WebSocketProxyTest.java [hunk 1]
```diff
@@ -148,6 +150,35 @@ protected PasswordAuthentication getPasswordAuthentication() {
...
+
+    static List<bytes> ofBytes(List<byte[]> bytes) {
+        return bytes.stream().map(bytes::new).toList();
+    }
+
+    static String diagnose(List<byte[]> a, List<byte[]> b) {
+        var actual = ofBytes(a);
+        var expected = ofBytes(b);
+        var problem = actual.equals(expected) ? "match" : "differ";
+        return "%s and %s %s".formatted(actual, expected, problem);
```
Comment (pavelrappo) [inline test/jdk/java/net/httpclient/websocket/WebSocketProxyTest.java:179]: I would rename "problem" to more neutral "message" or "description". Ditto in the other file.

**Review by pavelrappo — APPROVED** [review]

File: test/jdk/java/net/httpclient/websocket/WebSocketProxyTest.java (hunk 1 above)
Comment (dfuch) [reply test/jdk/java/net/httpclient/websocket/WebSocketProxyTest.java]: Done.

**Pull request closed without merging**